### PR TITLE
Azure-ci-hook

### PR DESCRIPTION
**Pull request by violetbrina**
self-explanatory

### CODE REVIEW

File: .github/workflows/prod_deploy.yaml [hunk 1]
```diff
@@ -7,18 +7,27 @@ jobs:
...
               -H "Content-Type:application/json" \
               -d '{"steps": ["deploy_auth", "deploy_batch", "deploy_ci", "deploy_hailgenetics_image", "deploy_memory", "deploy_wheel", "upload_query_jar"], "sha": "${{ github.sha }}"}' \
               https://ci.hail.populationgenomics.org.au/api/v1alpha/prod_deploy)
-          echo DEPLOY_BATCH_URL=$DEPLOY_BATCH_URL >> $GITHUB_ENV
+          echo DEPLOY_BATCH_URL=$DEPLOY_BATCH_URL_GCP >> $GITHUB_ENV
+
+      - name: deploy_azure
+        run: |
+          DEPLOY_BATCH_URL=$(curl --fail --silent --show-error -X POST \
+              -H "Authorization: Bearer ${{ secrets.CI_TOKEN }}" \
```
Comment (illusional) [inline .github/workflows/prod_deploy.yaml:22]: You'll need to come up with a different CI token I think, I think should be just grabbing ci token for the auto created user in kubes.

File: .github/workflows/prod_deploy.yaml [hunk 2]
```diff
@@ -7,18 +7,27 @@ jobs:
...
+              -H "Content-Type:application/json" \
+              -d '{"steps": ["deploy_auth", "deploy_batch", "deploy_ci"], "sha": "${{ github.sha }}"}' \
+              https://ci.azhail.popgen.rocks/api/v1alpha/prod_deploy)
+          echo DEPLOY_BATCH_URL=$DEPLOY_BATCH_URL_AZURE >> $GITHUB_ENV
```
Comment (illusional) [inline .github/workflows/prod_deploy.yaml:26]: I think this syntax should look like (and vice-versa for the above):

```suggestion
          echo DEPLOY_BATCH_URL_AZURE=$DEPLOY_BATCH_URL >> $GITHUB_ENV
```

**Review by illusional — CHANGES_REQUESTED** [review]

File: .github/workflows/prod_deploy.yaml [hunk 3]
```diff
@@ -7,18 +7,27 @@ jobs:
...
               https://ci.hail.populationgenomics.org.au/api/v1alpha/prod_deploy)
-          echo DEPLOY_BATCH_URL=$DEPLOY_BATCH_URL >> $GITHUB_ENV
+          echo DEPLOY_BATCH_URL_GCP=$DEPLOY_BATCH_URL >> $GITHUB_ENV
+
+      - name: deploy_azure
+        run: |
+          DEPLOY_BATCH_URL=$(curl --fail --silent --show-error -X POST \
+              -H "Authorization: Bearer ${{ secrets.CI_TOKEN_AZURE }}" \
+              -H "Content-Type:application/json" \
+              -d '{"steps": ["deploy_auth", "deploy_batch", "deploy_ci"], "sha": "${{ github.sha }}"}' \
```
Comment (lgruen) [inline .github/workflows/prod_deploy.yaml:24]: Why only those few steps here? We'll definitely need the Query JAR, for example, to run Query on Batch.

Can we share the same templated invocation and just interpolate the token + URL, so we don't run the risk of getting out-of-sync between GCP and Azure?

File: .github/workflows/prod_deploy.yaml [hunk 4]
```diff
@@ -7,18 +7,27 @@ jobs:
...
+              https://ci.azhail.popgen.rocks/api/v1alpha/prod_deploy)
+          echo DEPLOY_BATCH_URL_AZURE=$DEPLOY_BATCH_URL >> $GITHUB_ENV
 
       - name: post to Slack
         run: |
           curl --fail --silent --show-error -X POST \
               -H "Content-type: application/json" \
-              -d "{\"text\": \"Deploying Hail Batch: $DEPLOY_BATCH_URL\"}" \
+              -d "{\"text\": \"Deploying Hail Batch: $DEPLOY_BATCH_URL_GCP $DEPLOY_BATCH_URL_AZURE\"}" \
```
Comment (lgruen) [inline .github/workflows/prod_deploy.yaml:32]: Can we put the two URLs on separate lines and make really clear what refers to GCP vs Azure?

**Review by lgruen — CHANGES_REQUESTED** [review]

File: .github/workflows/prod_deploy.yaml (hunk 3 above)
Comment (violetbrina) [reply .github/workflows/prod_deploy.yaml:24]: Currently on Azure we have only the auth, batch and ci deployed. Running the ci deploy command  with all of them failed on `deploy_hailgenetics_image`. To be fair I hadn't checked the others but I haven't even tested them yet.

The current azure instructions stop at `deploy_batch`. I don't even know if the rest are implemented yet or not but I can ask.

File: .github/workflows/prod_deploy.yaml (hunk 4 above)
Comment (violetbrina) [reply .github/workflows/prod_deploy.yaml:32]: Sure!

File: .github/workflows/prod_deploy.yaml (hunk 3 above)
Comment (violetbrina) [reply .github/workflows/prod_deploy.yaml]: Ok, I've confirmed that `deploy_hailgenetics_image` and `deploy_wheel` are not implemented. The others I've successfully run and added to the ci hook as well.

Comment (lgruen) [reply .github/workflows/prod_deploy.yaml]: Thanks!

File: .github/workflows/prod_deploy.yaml [hunk 5]
```diff
@@ -7,18 +7,28 @@ jobs:
...
+              -d '{"steps": ["deploy_auth", "deploy_batch", "deploy_ci", "deploy_memory", "upload_query_jar"], "sha": "${{ github.sha }}"}' \
+              https://ci.azhail.popgen.rocks/api/v1alpha/prod_deploy)
+          echo DEPLOY_BATCH_URL_AZURE=$DEPLOY_BATCH_URL >> $GITHUB_ENV
 
       - name: post to Slack
         run: |
           curl --fail --silent --show-error -X POST \
               -H "Content-type: application/json" \
-              -d "{\"text\": \"Deploying Hail Batch: $DEPLOY_BATCH_URL\"}" \
+              -d "{\"text\": \"Deploying Hail Batch: \n( GCP ) $DEPLOY_BATCH_URL_GCP\n(Azure) $DEPLOY_BATCH_URL_AZURE\"}" \
```
Comment (lgruen) [inline .github/workflows/prod_deploy.yaml:33]: Assuming that formatting works here (might be worth testing whether the newlines work as well):

```suggestion
              -d "{\"text\": \"Deploying Hail Batch: \n_GCP:_ $DEPLOY_BATCH_URL_GCP\n_Azure:_ $DEPLOY_BATCH_URL_AZURE\"}" \
```

File: .github/workflows/prod_deploy.yaml [hunk 6]
```diff
@@ -7,18 +7,28 @@ jobs:
   invoke-prod-deploy:
     runs-on: ubuntu-latest
     steps:
-      - name: deploy
+      - name: deploy_gcp
```
Comment (lgruen) [inline .github/workflows/prod_deploy.yaml:10]:
```suggestion
      - name: deploy to GCP
```

File: .github/workflows/prod_deploy.yaml [hunk 7]
```diff
@@ -7,18 +7,28 @@ jobs:
...
         run: |
           DEPLOY_BATCH_URL=$(curl --fail --silent --show-error -X POST \
               -H "Authorization: Bearer ${{ secrets.CI_TOKEN }}" \
               -H "Content-Type:application/json" \
               -d '{"steps": ["deploy_auth", "deploy_batch", "deploy_ci", "deploy_hailgenetics_image", "deploy_memory", "deploy_wheel", "upload_query_jar"], "sha": "${{ github.sha }}"}' \
               https://ci.hail.populationgenomics.org.au/api/v1alpha/prod_deploy)
-          echo DEPLOY_BATCH_URL=$DEPLOY_BATCH_URL >> $GITHUB_ENV
+          echo DEPLOY_BATCH_URL_GCP=$DEPLOY_BATCH_URL >> $GITHUB_ENV
+
+      - name: deploy_azure
```
Comment (lgruen) [inline .github/workflows/prod_deploy.yaml:19]:
```suggestion
      - name: deploy to Azure
```

**Review by lgruen — APPROVED** [review]

**Review by illusional — APPROVED** [review]

**Pull request merged**